hydroflow_lang/src/graph/serde_graph.rs

```diff
@@ -0,0 +1,174 @@
+use slotmap::{Key, SecondaryMap, SlotMap, SparseSecondaryMap};
+
+use serde::{Deserialize, Serialize};
+
+use super::{GraphNodeId, GraphSubgraphId};
+
+#[derive(Default, Serialize, Deserialize)]
+#[allow(dead_code)] // TODO(mingwei): remove when no longer needed.
+pub struct SerdeGraph {
+    pub nodes: SecondaryMap<GraphNodeId, String>,
+    pub edges: SecondaryMap<GraphNodeId, Vec<GraphNodeId>>,
```

Comment:
Are the order of the edges preserved from PartitionedGraph?

Reply:
I guess they are, though this structure does lose the order information for predecessor edges.

Reply:
Oh that's interesting. I'll put in a comment, we should fix.